proxy-robots-module/src/main/java/orwell/proxy/IRobot.java

```diff
@@ -0,0 +1,63 @@
+package orwell.proxy;
+
+import orwell.messages.Controller.Input;
+
+public interface IRobot {
+
+    String getTeamName();
+
+    EnumRegistrationState getRegistrationState();
+
+    EnumConnectionState getConnectionState();
+
+    String getRoutingID();
+
+    void setRoutingID(String routingID);
+
+    String getImage();
+
+    /*
+     * Image is a picture of the robot itself to be sent to the
+     * server before the start of the game
+     */
+    void setImage(String image);
+
+    /*
+     * This will clear the current ServerRobotState
+     */
+    byte[] getAndClearZmqServerRobotStateBytes();
+
+    void setRegistered(byte[] registeredMessage);
+
+    byte[] getZmqRegister();
```

Comment:
The naming convention might not be very consistent with the following method (getRegisterBytes).
It might be better to state in the method that it is about a message. It is difficult to find good names ; today I would be inclined to suggest getMessageRegisterZmq / getMessageRegisterBytes.